tests/test_decoding.py

```diff
@@ -287,7 +288,10 @@ def test_decode_request_coded_const(self):
             ecu_variant_patterns=[],
         )
         diag_layer = DiagLayer(diag_layer_raw=diag_layer_raw)
-        diag_layer.finalize_init(odxlinks=odxlinks)
+        odxlinks = OdxLinkDatabase()
+        odxlinks.update(diag_layer._build_odxlinks())
+        diag_layer._resolve_odxlinks(odxlinks)
+        diag_layer._finalize_init(odxlinks)
```

Comment:
Not possible to merge _resolve_odxlinks and _finalize_init if both will always be called together?

Reply:
my plan is to get rid of the `odxlinks` argument for `finalize_init` in a follow-up PR, but that would have led to even more extensive changes here...

Reply:
I forgot to actual answer: `_finalize_init()` conceptually has not much to do with the ODXLINK mechanism (the current need for the `odxlinks` argument for `finalize_init()` is unfortunate, see comment above).

tl;dr: it's possible but its IMO not a terribly good idea.